stetho/src/main/java/com/facebook/stetho/inspector/protocol/module/DOM.java

```diff
@@ -172,64 +208,196 @@ public void run() {
     });
   }
 
-  private Node createNodeForElement(Object element) {
+  private void updateTree() {
+    long startTimeMs = SystemClock.elapsedRealtime();
+
+    ShadowDOM.Update domUpdate = createShadowDOMUpdate();
+    boolean isEmpty = domUpdate.isEmpty();
+    if (isEmpty) {
+      domUpdate.abandon();
+    } else {
+      applyDOMUpdate(domUpdate);
+    }
+
+    long endTimeMs = SystemClock.elapsedRealtime();
+    long deltaMs = endTimeMs - startTimeMs;
```

Comment:
nit: you can simplify this pattern with:

```java
long deltaMs = SystemClock.elapseRealtime() - startTimeMs;
```

It's quite rare that you actually care about the end time.